crates/task-impls/src/quorum_proposal.rs

```diff
@@ -0,0 +1,541 @@
+                    ProposalDependency::QC => {
+                        if let HotShotEvent::QCFormed(either::Left(qc)) = event {
+                            qc.view_number
+                        } else {
+                            return false;
+                        }
+                    }
+                    ProposalDependency::TimeoutCert => {
+                        if let HotShotEvent::QCFormed(either::Right(timeout)) = event {
+                            timeout.view_number
+                        } else {
+                            return false;
+                        }
+                    }
+                    ProposalDependency::ViewSyncCert => {
+                        if let HotShotEvent::ViewSyncFinalizeCertificate2Recv(view_sync_cert) =
+                            event
+                        {
+                            view_sync_cert.view_number
+                        } else {
+                            return false;
+                        }
+                    }
+
+                    ProposalDependency::ProposalCertificate => {
+                        if let HotShotEvent::QuorumProposalRecv(proposal, _) = event {
+                            proposal.data.view_number
+                        } else {
+                            return false;
+                        }
+                    }
+                    ProposalDependency::PayloadAndMetadata => {
+                        if let HotShotEvent::SendPayloadCommitmentAndMetadata(
+                            _payload_commitment,
+                            _metadata,
+                            view,
+                        ) = event
+                        {
+                            *view
+                        } else {
+                            return false;
+                        }
+                    }
+                };
+                event_view == view_number
+            }),
+        )
+    }
+
+    /// Create and store an [`AndDependency`] combining [`EventDependency`]s associated with the
+    /// given view number if it doesn't exist. Also takes in the received `event` to seed a
+    /// dependency as already completed. This allows for the task to receive a proposable event
+    /// without losing the data that it received, as the dependency task would otherwise have no
+    /// ability to receive the event and, thus, would never propose.
+    fn create_dependency_task_if_new(
+        &mut self,
+        view_number: TYPES::Time,
+        event_receiver: Receiver<Arc<HotShotEvent<TYPES>>>,
+        event_sender: Sender<Arc<HotShotEvent<TYPES>>>,
+        event: Arc<HotShotEvent<TYPES>>,
+    ) {
+        debug!("Attempting to make dependency task for event {:?}", event);
+        if self.propose_dependencies.get(&view_number).is_some() {
+            debug!("Task already exists");
+            return;
+        }
+
+        let mut proposal_cert_validated_dependency = self.create_event_dependency(
```

Comment:
Nit: similar to my previous comment, I think it's better to remove `cert` from the name.

Reply:
Done.